Library/Homebrew/livecheck/options.rb

```diff
@@ -0,0 +1,105 @@
+# typed: strict
+# frozen_string_literal: true
+
+module Homebrew
+  module Livecheck
+    # Options to modify livecheck's behavior. These primarily come from
+    # `livecheck` blocks but they can also be set by livecheck at runtime.
+    class Options
+      # Whether to use brewed curl.
+      sig { returns(T.nilable(T::Boolean)) }
+      attr_reader :homebrew_curl
+
+      # Form data to use when making a `POST` request.
+      sig { returns(T.nilable(T::Hash[T.any(String, Symbol), String])) }
+      attr_reader :post_form
+
+      # JSON data to use when making a `POST` request.
+      sig { returns(T.nilable(T::Hash[T.any(String, Symbol), String])) }
+      attr_reader :post_json
+
+      # @param homebrew_curl whether to use brewed curl
+      # @param post_form form data to use when making a `POST` request
+      # @param post_json JSON data to use when making a `POST` request
+      sig {
+        params(
+          homebrew_curl: T.nilable(T::Boolean),
+          post_form:     T.nilable(T::Hash[T.any(String, Symbol), String]),
+          post_json:     T.nilable(T::Hash[T.any(String, Symbol), String]),
+        ).void
+      }
+      def initialize(homebrew_curl: nil, post_form: nil, post_json: nil)
+        @homebrew_curl = homebrew_curl
+        @post_form = post_form
+        @post_json = post_json
+      end
+
+      # Returns a `Hash` of options that are provided as arguments to `url`.
+      sig { returns(T::Hash[Symbol, T.untyped]) }
+      def url_options
+        {
+          homebrew_curl: @homebrew_curl,
+          post_form:     @post_form,
+          post_json:     @post_json,
+        }
+      end
+
+      # Returns a `Hash` of all instance variable values.
+      sig { returns(T::Hash[Symbol, T.untyped]) }
+      def to_h
+        {
+          homebrew_curl: @homebrew_curl,
+          post_form:     @post_form,
+          post_json:     @post_json,
+        }
+      end
```

Comment:
```suggestion
      alias to_h url_options
```

Reply:
`to_h` and `url_options` are strictly aligned at the moment but they won't stay the same forever. I plan to add strategy options in the future (for modifying strategy behavior), so I specifically split out `url_options` as a way of categorizing options. It looks redundant right now but it will make sense later.